lib/awful/client.lua

```diff
@@ -634,6 +634,24 @@ function client.object.is_fixed(c)
     return false
 end
 
+--- Is the client immobilized, i.e. fullscreen or maximized?
+--
+-- **Signal:**
+--
+--  * *property::is_immobilized*
+--
+-- This property is read only.
+-- @property is_immobilized
+-- @param boolean The immobilized state
+-- @see maximized
+-- @see maximized_horizontal
+-- @see maximized_vertical
+-- @see fullscreen
+
+function client.object.is_immobilized(c)
+    return c.fullscreen or c.maximized or c.maximized_horizontal or c.maximized_vertical
```

Comment:
Horizontally maximized clients should only be horizontally immobilized, I think. So, this needs to be split up into two properties?